File: 151004/Belsky/rv-lab1/src/main/java/com/example/rvlab1/model/entity/UserEntity.java
package com.example.rvlab1.model.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
@Entity
@Table(name = "tblUser")
public class UserEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @Column(name = "login", length = 64, nullable = false, unique = true)
    private String login;

    @Column(name = "password", length = 128, nullable = false)
    private String password;

    @Column(name = "firstname", length = 64, nullable = false)
    private String firstname;

    @Column(name = "lastname", length = 64, nullable = false)
    private String lastname;
}

File: 251002/Belyatskiy/2222/lab1/src/entities/user.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from 'typeorm';
import { Article } from './article.entity.js';

@Entity('tbl_user')
export class User {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ length: 64,type: 'varchar' })
    login: string;

    @Column({ length: 128,type: 'varchar' })
    password: string;

    @Column({ length: 64,type: 'varchar' })
    firstname: string;

    @Column({ length: 64,type: 'varchar' })
    lastname: string;

    @OneToMany(() => Article, (article) => article.user)
    articles: Article[];
}